satpy/readers/__init__.py

```diff
@@ -231,6 +244,61 @@ def _get_sorted_file_groups(all_file_keys, time_threshold):
     return file_groups
 
 
+def _filter_groups(groups, missing="pass"):
+    """Filter multi-reader group-files behavior.
+
+    Helper for `group_files`.  When `group_files` is called with multiple
+    readers, make sure that the desired behaviour for missing files is
+    enforced: if missing is ``"raise"``, raise an exception if at least one
+    group has at least one reader without files; if it is ``"skip"``, remove
+    those.  If it is ``"pass"``, do nothing.
+
+    Args:
+        groups (List[Mapping[str, List[str]]]):
+            groups as found by `group_files`.
+        missing (str):
+            String controlling behaviour, see documentation above.
+
+    Returns:
+        None (changes groups in-situ)
+    """
+    if missing == "pass":
+        return
+    elif missing in ("raise", "skip"):
```

Comment:
You could change this to `if missing not in ("raise", "skip"): raise ValueError(...)` and then dedent the rest of the function.

Reply:
Done.